### ISSUE TITLE
Update libsass-net to latest libsass (3.1)

### ISSUE THREAD

**Issue by RobertjanTuit** [darrenkopp/libsass-net#13]
Hi Darren,

Really appreciate the work you have done on libsass-net. It has been a life saver in our project.

But as of late I can no longer get it to work with the latest libsass master. Would you be able to give me some hints into how I would get this to run?

We would love to update libsass to 3.1 with the new sass functionalities that come with it.

Cheers,
  _Rj

**Comment by darrenkopp** [comment]
Yeah, right now there isn't a good way to get the latest. I've been working
with the libsass team and they are working on publishing an official
libsass DLL for windows which I will then update libsass-net to use but it
isn't out yet.

I can do it before then, it'd just be duplicate work. Once we do pinvoke vs
CLR c++ we solve a couple of other issues in this project as well.

**Comment by RobertjanTuit** [comment]
Awesome, thanks for the amazingly quick reply ☺

Do you have any idea on a timeline, and/or is there any way I can help/support to bring the timeline in?

_Rj

**Comment by darrenkopp** [comment]
Well, mostly waiting on the official dll's to be released, so likely whenever libsass ends up doing another release. Then the only work is to update libsass to use p/invoke and use their updated api.

I could re-create what libsass has already done with the official windows driver, but there isn't a guarantee that it will 100% work with 3.1 tag. I'll ping the libsass team and see where they are on the official driver.

**Comment by am11** [comment]
@darrenkopp, we are building windows binary in node-sass for v3.1 with VS2013. Can you share from that?

Also, libsass now has VS solution and project files in win directory: https://github.com/sass/libsass/tree/master/win, which emits the dll and (sassc's) exe. If you want only dll, you can replace `libsass.vcxproj` in their folder with this one: https://github.com/am11/libsass/blob/013e98bdd71fbec7beab84a31bddae8c97120147/win/libsass.vcxproj, launch the `libsass.sln` in VS2013, Ctrl+Shift+B and done! :)

AFAICT, they are planning on releasing official binary for Redhat and probably for other OSes as well. But that might not happen anytime soon (until after v3.2? //cc @mgreter).

Is there anything we can help moving things forward here?

**Comment by mgreter** [comment]
I don't think libsass will ever release compiled binaries/libraries anytime! We only try to make it as easy for package maintainers of distributions to create releases. Or what made you think so?

**Comment by am11** [comment]
@mgreter, 

I think the news about official RPM etc. might have be misinterpreted.

On a related note, if we isolate libsass VS project file from sassc dependency (like https://github.com/sass/libsass/pull/810, a better version; with explicit sources and header like you suggested), projects like this can add our `.vcxproj` as is (and in case of .NET wrapper, the downstream maintainer can change `VS > (open solution) > Solution Explorer > (right-click libsass project) > Properties > Configuration Properties > Common Language Runtime Support = No Common Language Runtime Support` to dismiss CLR and go fully native. Then they can create another project under the same solution (say `binding`) for their [wrapper code](https://github.com/darrenkopp/libsass-net/tree/7f7abb3c98cf4df90c51dfda2490278601fd1797/libsass) (which depends on CLR to support .NET runtime) and reference libsass project as is. This way they don't have to add new files to project every time libsass has new sources.

We can then probably make node-sass binding depend on libsass project file in a similar fashion and get rid off of [explicit inclusions](https://github.com/sass/node-sass/blob/b04137d0adc54c7e8fe7f1d4a2eebfaa646dd6b7/binding.gyp#L6-L41) (at least for windows). I will do some more digging on how to make it happen with gyp (which emits its own VS solution and project file). Down the road, we can probably use GYP or CMake in libsass to automate the whole process. Build systems are not my strong suits, but no harm in giving it a shot. :smiley: 

@darrenkopp, for v3.1.0, I think c87a756  via #14 will do the job. I suggest to implement new libsass API and to get richer experience and much more features and fine-grained configuration options! I you agree on that, I will give it a try if I can pull it off. FTR, I have never used libsass-net in any real world scenario, so I can't envision how the end-user can consume extra-ordinary cool features like Custom Importers and Custom Functions, but I am certainly willing to take a shot at it as well (if those are appropriate and have room for the consumers of libsass-net; human or software: Sassy Studio, MSBuild tasks etc.). :sunglasses: 

Separately, should we bump .NET Framework version to 4.5 in libsass solution and then [analyze if](https://github.com/dotnet/corefx/issues/973#issuecomment-76052684) the library is also compatible with .NET Core Framework?

**Comment by darrenkopp** [comment]
@am11 :+1: will look into it, thanks

**Comment by darrenkopp** [comment]
available on nuget now.

**Comment by RobertjanTuit** [comment]
Thanks Darren!

Sent from my iPhone

On Mar 23, 2015, at 9:45 AM, Darren Kopp <notifications@github.com<mailto:notifications@github.com>> wrote:


available on nuget now.

**Comment by samrueby** [comment]
Thanks!

**Comment by am11** [comment]
:+1:
